openhwgroup/core-v-verif
# Update cv32e40s csr access test

What this PR does:
- update gen_csr_access_test script:
1) make it possible to specify input path for the yaml file
2) make sure we use the m4 yaml file if the m4 flag is on
3) update help/args description 

- cv32e40s_csr_access_test:
1) update the file by using an updated m4 yaml file (https://github.com/openhwgroup/cv32e40s/blob/master/yaml/csr.yaml.m4, will make a PR for this in the cv32e40s repo)

- test.yaml
1) make sure we dont let the compiler optimize the code. The optimization changes the bne instructions to a cm.jt, and consequently generates an infinit loop.

Extra:
Tried rebasing, which almost work (but eventually ended up with a mess). So unless anyone complains about the commits, I will let it be.

This PR is related to another PR in the cv32e40x repo: https://github.com/openhwgroup/cv32e40x/pull/853 (which initially was in the *s repo (https://github.com/openhwgroup/cv32e40s/pull/453))
